### DISCUSSION TITLE
JHTML Calendar min and max date select

### DISCUSSION THREAD

**Discussion by antonzeno** [joomla/joomla-cms#47138] [2. Cold Features]
Hello, Is there any way I can limit the min and max date selection on the calendar control field ? I  need to limit it in a max selection of 6 months from today date but couldnt find any info regarding it. Any help would be appreciated. Thanks

**Comment by antonzeno** [comment]
> Range selection in a date range picker can be made-to-order with desire restrictions based on the application needs.
> We can set by using starting date and ending date .

Is there any link to doc for this? I cant find it anywhere

**Comment by antonzeno** [comment]
The only solution I could find for now is using the minYear and maxYear attributes. But it doesnt seem to solve the issue as I need to put a more restricted limit, and not limit it to a specific year.

**Comment by ReLater** [comment]
> Is there any link to doc for this? I cant find it anywhere

There's no doc for your issue (https://docs.joomla.org/Calendar_form_field_type). I think shub1711 is just a troll.

> The only solution I could find for now is using the minYear and maxYear attributes

You are right. There is no other setting in Joomla 3 or Joomla 4 (as far as I see after code investigations). You can try to make a feature request out of this issue for Joomla 4. 

Joomla 3 doesn't accept new features.

**Comment by antonzeno** [comment]
> You are right. There is no other setting in Joomla 3 or Joomla 4 (as far as I see after code investigations). You can try to make a feature request out of this issue for Joomla 4.
> 
> Joomla 3 doesn't accept new features.

Thanks, I saw there has been a PR a few years ago to add dateStart and dateEnd but it looks like those never got implemented

**Comment by rs4231199** [comment]
@AndonZeno Can you share the link to the PR?

**Comment by antonzeno** [comment]
> https://ej2.syncfusion.com/documentation/daterangepicker/range-restriction/#:~:text=You%20can%20restrict%20the%20minimum,can%20be%20selected%20as%20endDate.
> 
> May be it will help.

Its not relevant to the question.

> @AndonZeno Can you share the link to the PR?

https://github.com/joomla/joomla-cms/pull/4814

**Comment by rs4231199** [comment]
So, the PR was closed because the author never responded. I am working on this now.

update: I am no longer working on it.

**Comment by brianteeman** [comment]
Can someone please add the J4 Issue and Feature Request labels please

**Comment by jjnxpct** [comment]
I was looking for this option/feature for one of our J4 websites. I think this has not been added yet? Is there anyone who can add the min and max date attributes to the calendar field? I would do it myself if I could, but unfortunately don't know how.

**Comment by antonzeno** [comment]
> I was looking for this option/feature for one of our J4 websites. I think this has not been added yet? Is there anyone who can add the min and max date attributes to the calendar field? I would do it myself if I could, but unfortunately don't know how.

Im working on it will do a PR soon

**Comment by TLWebdesign** [comment]
@antonzeno how far have you gotten?
